File: Java/JavaUtils/src/javautils/message/IMessageHeader.java
package javautils.message;

import java.io.DataInputStream;
import java.io.DataOutputStream;

public interface IMessageHeader
{
    /***************************************************************************
     * Returns a hash of the sender.
     * 
     * @return
     **************************************************************************/
    public int getSenderHash();

    /***************************************************************************
     * Sets the hash of the sender.
     * 
     * @param hash
     **************************************************************************/
    public void setSenderHash( int hash );

    /***************************************************************************
     * Returns the message type.
     * 
     * @return
     **************************************************************************/
    public MessageType getMessageType();

    /***************************************************************************
     * Sets the message type.
     * 
     * @param type
     **************************************************************************/
    public void setMessageType( MessageType type );

    /***************************************************************************
     * Returns the message length (including the length of the header).
     * 
     * @return
     **************************************************************************/
    public long getMessageLength();

    /***************************************************************************
     * Sets the message length.
     * 
     * @param length
     **************************************************************************/
    public void setMessageLength( long length );

    /***************************************************************************
     * Writes this header to a {@link DataOutputStream}.
     * 
     * @param stream
     * @throws Exception
     **************************************************************************/
    public void headerToBinaryStream( DataOutputStream stream )
            throws Exception;

    /***************************************************************************
     * Reads this message from the supplied {@link DataInputStream}.
     * 
     * @param stream
     * @return
     * @throws Exception
     **************************************************************************/
    public IMessageHeader binaryStreamToHeader( DataInputStream stream )
            throws Exception;
}

File: C#/ChatterBox/ChatterBox/IMessageHeader.cs
using System.IO;

namespace ChatterBox
{
    interface IMessageHeader
    {
        /// <summary>
        /// Returns the hash code representing the sender of this message.
        /// </summary>
        /// <returns></returns>
        int getSenderHash();

        /// <summary>
        /// Sets the hash code representing the sender of this message.
        /// </summary>
        /// <param name="hash"></param>
        void setSenderHash( int hash );

        /// <summary>
        /// Returns the type of message.
        /// </summary>
        /// <returns></returns>
        MessageUtils.MessageType getMessageType();

        /// <summary>
        /// Sets the type of message.
        /// </summary>
        /// <param name="type"></param>
        void setMessageType( MessageUtils.MessageType type );

        /// <summary>
        /// Returns this message's length.
        /// </summary>
        /// <returns></returns>
        int getMessageLength();

        /// <summary>
        /// Sets this message's length.
        /// </summary>
        /// <param name="length"></param>
        void setMessageLength( int length );

        /// <summary>
        /// Writes this message header to the supplied BinaryWriter.
        /// </summary>
        /// <param name="writer"></param>
        void headerToBinaryStream( BinaryWriter writer );

        /// <summary>
        /// Reads a message header from the supplied BinaryReader.
        /// </summary>
        /// <param name="reader"></param>
        /// <returns></returns>
        IMessageHeader headerFromBinaryStream( BinaryReader reader );
    }
}
